planning/louketo-proxy-plans.md

```diff
@@ -0,0 +1,420 @@
+
+    * If we need custom extensions it may be complicated to maintain and deliver a separate distribution
+
+## Pusher OAuth2 Proxy
+
+[https://pusher.github.io/oauth2_proxy/](https://pusher.github.io/oauth2_proxy/)
+
+[https://github.com/pusher/oauth2_proxy](https://github.com/pusher/oauth2_proxy) 
+
+### Overview
+
+Pusher OAuth2 proxy is a simple proxy that provides authentication using an identity provider. There is support for a few providers built-in, including Google, Azure, Facebook, Keycloak and generic OpenID Connect. Configuration and documentation is a bit confusing due to a lot of provider specific content, and not that much general content. It looks like it’s only covering the use-case of authentication for a web application, and it seems there is no support for authenticating incoming requests using bearer tokens. There is no authorization support.
+
+### Project
+
+* Medium community (1.2k stars, 138 contributors)
+
+* MIT license
+
+* Implemented in Go
+
+* Fork from [https://github.com/bitly/oauth2_proxy](https://github.com/bitly/oauth2_proxy)
+
+### Notes
+
+* Supports authentication for server-side web application
+
+* A lot of needed features are missing
+
+    * No support for services
+
+    * No support for authorization
+
+    * No support for plugins/extensions
+
+    * No support for logout
+
+* Big focus on social providers, which is useful, but at the same time distracting as there is a lot of information in docs and config that is provider specific
+
+* Documentation is okay, but a lot of focus on specific providers
+
+* Doesn’t look like there is a quickstart/example
+
+* Can forward tokens, but not custom claims to application
+
+* Supports a number of social providers
+
+* Supports OpenID Connect providers
+
+    * Loads keys from JWK (but probably doesn’t support key rotation)
+
+* Sessions via cookies only, supports encryption
+
+* Not sure if it adds any Security Headers to responses
+
+* Doesn’t look like there is CSRF protection
+
+* Container image size: 10.9 MB
+
+## Keycloak Gatekeeper
+
+[https://github.com/keycloak/keycloak-gatekeeper](https://github.com/keycloak/keycloak-gatekeeper)
+
+[https://www.keycloak.org/docs/latest/securing_apps/index.html#_keycloak_generic_adapter](https://www.keycloak.org/docs/latest/securing_apps/index.html#_keycloak_generic_adapter)
+
+### Overview
+
+Gatekeeper is a proxy aiming to secure existing and new applications and services. It supports authenticating users through an OpenID Connect provider. It also supports securing services with bearer tokens. There is good support for defining basic authorization policies based requiring claims for specific paths. There is a big focus on Keycloak even though it is generic and could be used with any OpenID Connect provider. 
+
+### Project
+
+    * Small community (593 stars, 32 contributors)
+
+    * ASL2 license
+
+    * Implemented in Go
+
+### Notes
```

Comment:
@stianst maybe we can convert some of these items into milestones? Once we agree on each milestone we can start creating new issues.

Reply:
+1